### PR TITLE
Initial docs for distgen

### PR DESCRIPTION
**Pull request by bkabrda**
This is the initial version of docs for distgen. When this PR merged, I'll register distgen at readthedocs.org, create documentation there and then put a link to README to point there.

### CODE REVIEW

**Comment by bkabrda** [comment]
Fixes #14

File: docs/intro.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+Download/Installation
+---------------------
+
+Stable releases of distgen are available (for Red Hat distributions) in
+`Copr <https://copr.fedoraproject.org/coprs/praiskup/distgen>`_.
+
+You can also run development version directly from
+`Github <https://github.com/devexp-db/distgen/>`_, simply use the
+``dg`` shell wrapper available in git root directory.
+
```
Comment (eliskasl) [inline docs/intro.rst]: ...which fails with a traceback if you don't have the yaml module installed. Maybe worth mentioning...? If not here, in the README at least perhaps

File: docs/intro.rst [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+         {{ commands.pkginstaller.cleancache() }}
+     
+     RUN echo '#!/bin/bash' > /usr/bin/script.sh && \
+         echo "exec nc -kl 9000 -c 'echo -e \"HTTP/1.1 200 OK\n\";echo \"I am awesome\"'" >> /usr/bin/script.sh && \
+         chmod +x /usr/bin/script.sh
+     
+     CMD ["/usr/bin/script.sh"]
+
+
+4. Run ``dg`` tool to generate **Fedora 26** Dockerfile with software
```
Comment (eliskasl) [inline docs/intro.rst]: Run the... for a...

File: docs/intro.rst [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+     # when using "--multispec", "--multispec-selector" must be used for all
+     # spec groups except "distroinfo"
+     $ dg --template Dockerfile.template \
+          --spec common.yaml \
+          --multispec multispec.yaml \
+          --multispec-selector version=2.4 \
+          --distro fedora-26-x86_64.yaml \
+     > Dockerfile
+
+5. Run ``dg`` tool again to generate **CentOS 7** dockerfile with software
```
Comment (eliskasl) [inline docs/intro.rst]: Run the... for a... Dockerfile

File: docs/intro.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+     $ dg --template Dockerfile.template \
+          --spec common.yaml \
+          --multispec multispec.yaml \
+          --multispec-selector version=2.2 \
+          --distro centos-7-x86_64.yaml \
+     > Dockerfile
+
+There are more nuances and features of distgen that you can utilize,
+all of them are documented in following sections of this documentation.
```
Comment (eliskasl) [inline docs/intro.rst]: in the following

File: docs/spec_multispec.rst [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+   version: 2.4
+
+Example template::
+
+   This is documentation for version {{ spec.version }} of some software.
+
+By using specs with different ``version`` in the example above, you could
+render the template for various software versions. While this is ok for
+simpler usecases, it might become impractical on bigger scale: imagine
+you want to render Dockerfile for an image, that will be based on several
```
Comment (eliskasl) [inline docs/spec_multispec.rst]: render a Dockerfile

File: docs/spec_multispec.rst [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+files, each with couple of lines and you'd need to manually select and
+pass them to the ``dg`` command. This is why the *multispec* mechanism
+was added to distgen.
+
+Multispecs
+----------
+
+Multispec is a file that solves two problems:
+
+* Merge several different smaller spec files into a single file for better
```
Comment (eliskasl) [inline docs/spec_multispec.rst]: Merges

File: docs/spec_multispec.rst [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  readability and convenience.
+* Put smaller specs in logical groups and define a "matrix" - a list of all
```
Comment (eliskasl) [inline docs/spec_multispec.rst]: Puts ... and defines

File: docs/intro.rst (hunk 1 above)
Comment (bkabrda) [reply docs/intro.rst]: Right, I should mention installation of dependencies - thanks!

**Comment by bkabrda** [comment]
Thanks for the review. I fixed all the issues that were pointed out, so I'm merging now.

**Pull request merged**